src/qibo/backends/__init__.py

```diff
@@ -78,39 +79,95 @@ class GlobalBackend(NumpyBackend):
+        return cls._backend
 
-        backend = os.environ.get("QIBO_BACKEND")
-        if backend:  # pragma: no cover
+    @classmethod
+    def _create_backend(cls):
+        backend_env = os.environ.get("QIBO_BACKEND")
+        if backend_env:  # pragma: no cover
             # Create backend specified by user
             platform = os.environ.get("QIBO_PLATFORM")
-            cls._instance = construct_backend(backend, platform=platform)
+            backend = construct_backend(backend_env, platform=platform)
         else:
             # Create backend according to default order
             for kwargs in cls._default_order:
                 try:
-                    cls._instance = construct_backend(**kwargs)
+                    backend = construct_backend(**kwargs)
                     break
-                except (ImportError, MissingBackend):
+                except (ModuleNotFoundError, ImportError):
                     pass
 
-        if cls._instance is None:  # pragma: no cover
+        if backend is None:  # pragma: no cover
             raise_error(RuntimeError, "No backends available.")
+        return backend
 
-        log.info(f"Using {cls._instance} backend on {cls._instance.device}")
-        return cls._instance
+    @classmethod
+    def set_backend(cls, backend, **kwargs):
+        cls._backend = construct_backend(backend, **kwargs)
+        log.info(f"Using {cls._backend} backend on {cls._backend.device}")
 
     @classmethod
-    def set_backend(cls, backend, **kwargs):  # pragma: no cover
-        if (
-            cls._instance is None
-            or cls._instance.name != backend
-            or cls._instance.platform != kwargs.get("platform")
-        ):
-            cls._instance = construct_backend(backend, **kwargs)
-        log.info(f"Using {cls._instance} backend on {cls._instance.device}")
+    def transpiler(cls):
+        """Get the current transpiler. If no transpiler is set, it will create one."""
+        if cls._transpiler is not None:
+            return cls._transpiler
+
+        cls._transpiler = cls._default_transpiler()
+        return cls._transpiler
+
+    @classmethod
+    def set_transpiler(cls, transpiler):
+        cls._transpiler = transpiler
+        # TODO: check if transpiler is valid on the backend
+
+    @classmethod
+    def _default_transpiler(cls):
+        import networkx as nx
+
+        from qibo.transpiler.optimizer import Preprocessing
+        from qibo.transpiler.pipeline import Passes
+        from qibo.transpiler.placer import Trivial
+        from qibo.transpiler.router import Sabre
+        from qibo.transpiler.unroller import NativeGates, Unroller
+
+        qubits: list[str] = list(cls._backend.qubits)  # list of qubit names
+        natives: list[str] = cls._backend.natives
+        connectivity_edges: list[tuple[str, str]] = (
+            cls._backend.connectivity
+        )  # list of edges
+        if qubits and natives and connectivity:
```

Comment:
```suggestion
        if qubits is not None and natives is not None and connectivity is not None:
```

Do not rely on `None` falsiness, check explicitly for it (`None` is different from `[]`, but they are both falsy).